Tests/Unit/MSFT_xIISFeatureDelegation.tests.ps1

```diff
@@ -4,14 +4,49 @@ $global:DSCResourceName = 'MSFT_xIISFeatureDelegation'
 #region HEADER
 
 [String] $moduleRoot = Split-Path -Parent (Split-Path -Parent (Split-Path -Parent $Script:MyInvocation.MyCommand.Path))
- if ( (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'DSCResource.Tests'))) -or `
-      (-not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'DSCResource.Tests\TestHelper.psm1'))) )
+$repoSource = (Get-Module -Name $global:DSCModuleName -ListAvailable)
+
+if (($repoSource -ne $null) -and ($repoSource[0].RepositorySourceLocation.Host -eq 'www.powershellgallery.com'))
+{
+    if ( -not (Test-Path -Path (Join-Path -Path $moduleRoot -ChildPath 'Tests\DscResourceTestHelper')) )
+    {
+
+        $choice = read-host "In order to run this test you need to install a helper module, continue with installation? (Y/N)"
```

Comment:
We can certainly add the switch to the test file, however not sure we can pass it to all files when executing tests via Invoke-Pester, so in those cases user still would be prompted. 
I'm not a big fan of prompts either, however not sure if customers would be fine to download the helper module when running tests in production, so it was just added as a warning to make them aware of what's happening.. I'm open to suggestions if you see alternative solutions though